internal/providers/gce/gce.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The gce provider fetches a remote configuration from the gce user-data
+// metadata service URL.
+
+package gce
+
+import (
+	"errors"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/coreos/ignition/config"
+	"github.com/coreos/ignition/config/types"
+	"github.com/coreos/ignition/internal/log"
+	"github.com/coreos/ignition/internal/providers"
+	putil "github.com/coreos/ignition/internal/providers/util"
+	"github.com/coreos/ignition/internal/util"
+)
+
+const (
+	initialBackoff = 100 * time.Millisecond
+	maxBackoff     = 30 * time.Second
+	baseUrl        = "http://metadata.google.internal/computeMetadata/v1/instance/"
+	userdataUrl    = baseUrl + "attributes/user-data"
+	sshKeysUrl     = baseUrl + "attributes/ssh-keys"
```

Comment:
I vote the first iteration of ignition+gce just does user-data and leaves the ssh key processing to the current script so we don't change behavior there just yet.